### DISCUSSION TITLE
bridge and exit

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [oxen-io/lokinet#1827] [Ideas]
> **Allow users in censored countries to use your network connection to access the Llokinet network.**

> **Help people in censored countries bypass censorship by running the Llokinet extension**

The lokinet (”bridge“and "exit") should be built into the lokinet browser and cannot be disabled.
It can run on PC: lokinet browser,
Android: lokinet browser
Other browsers. Provides extensions similar to [Tor Snowflake](https://chrome.google.com/webstore/detail/snowflake/mafpmfcccpbjnhfhjnllmmalhifmlcie)

[DDNS](https://en.wikipedia.org/wiki/Dynamic_DNS)   (”bridge“and "exit")

client discovers the IP address and port of nodes in several different ways.
1. Nodes discover their own external address by various methods.
2. Nodes receive the callback address of remote nodes that connect to them.
3. Nodes makes DNS request to receive IP addresses.
4. Nodes can use addresses hard coded into the software.
5. Nodes exchange addresses with other nodes.
6. Nodes store addresses in a database and read that database on startup.
7. Nodes can be provided addresses as command line arguments
8. Nodes read addresses from a user provided text file on startup

**Comment by jagerman** [comment]
What?

**Reply by majestrate** [reply]
i think he's talking about using user namespaces to do a lightweight container that forces everything over lokinet.
